### PR TITLE
MTM-56205 Improved description of how the inventory roles configurati…

### PR DESCRIPTION
**Pull request by pkepa7**
…on works

### CODE REVIEW

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 1]
```diff
@@ -47,9 +47,14 @@ Permissions are grouped into the following categories:
...
+Note that if a measurements also contains other fragments types, they also must be added in the inventory role configuration, otherwise, such measurements will not be returned because they contain fields to which the user has not been granted access.
+
+From version 10.17, an additional configuration was created for measurements (other categories remain unchanged). After setting the "acl.measurement.only-accessible-fragments" parameter to true, in order to have access to a single measurement parameter, we do not have to grant access to all fragments types that the measurement has. After switching the flag, only measurements that have a defined set of types will be returned, and additional type not listed in the inventory role configuration is removed from the returned measurements.
```
Comment (rssag) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:56]:
```suggestion
With version 10.17 an additional configuration was created for measurements (other categories remain unchanged). After setting the "acl.measurement.only-accessible-fragments" parameter to true, in order to have access to a single measurement parameter, you do not have to grant access to all fragments types that the measurement has. Only measurements that have a defined set of types are returned, and additional types not listed in the inventory role configuration are removed from the returned measurements.
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 2]
```diff
@@ -47,9 +47,14 @@ Permissions are grouped into the following categories:
...
+This will allow the user to see measurements that contain only defined types e.g. "c8y&#95;SignalStrength" without the additional need to configure other types of fragments that the measurement has.
```
Comment (rssag) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:57]:
```suggestion
This allows the user to see measurements that contain only the defined types, for example, "c8y_SignalStrength", without the additional need to configure other types of fragments that the measurement has.
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 3]
```diff
@@ -47,9 +47,14 @@ Permissions are grouped into the following categories:
 
 Add a permission to the role by clicking the plus icon next to the desired category.
 
-In the **Type** field, specify a type to further restrict the type of data that this permission applies to. Access will be only granted to objects that contain the specified **Type**.
+In the **Type** field, specify a type to further restrict the type of data that this permission applies to. Access will be only granted to objects that contain
+only the specified **Type**. If the selected object contains more fragment types than those defined in the inventory roles configuration, in order to display it, they also need to be added to the inventory role configuration.
```
[the remark this answers is not in the document]
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 4]
```diff
@@ -47,9 +47,14 @@ Permissions are grouped into the following categories:
...
 
-For example, assume that your device sends measurements related to device management, such as "c8y&#95;SignalStrength", and actual production measurements. You want a user to only see the device management measurements. In this case, enter "c8y&#95;SignalStrength" as type. This will allow the user to only see measurements that contain the type "c8y&#95;SignalStrength". Note that the user will be able to see the entire measurement object including other types that may be part of the same measurement object.
+For example, assume that your device sends measurements related to device management, such as "c8y&#95;SignalStrength" but measurement itself also has "c8y&#95;Temperature" in which we are not interested. You want a user to only see the device management measurements. In the default configuration for measurement, we must provide all types of fragments that measurement has i.e "c8y&#95;SignalStrength" and "c8y&#95;Temperature".
```
[the remark this answers is not in the document]
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 5]
```diff
@@ -47,9 +47,14 @@ Permissions are grouped into the following categories:
...
+Note that if a measurements also contains other fragments types, they also must be added in the inventory role configuration, otherwise, such measurements will not be returned because they contain fields to which the user has not been granted access.
```
[the remark this answers is not in the document]
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 1 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 2 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 6]
```diff
@@ -47,9 +47,14 @@ Permissions are grouped into the following categories:
...
+For example, assume that your device sends measurements related to device management, such as "c8y_SignalStrength" but the measurement itself also has "c8y_Temperature" which you are not interested in. You want a user to only see the device management measurements. In the default configuration for the measurement, provide all types of fragments that the measurement has, that is, "c8y_SignalStrength" and "c8y_Temperature".
+Note that if a measurement also contains other fragments types, they must also be added in the inventory role configuration. Otherwise such measurements are not returned because they contain fields to which the user has not been granted access.
+
+With version 10.17 an additional configuration was created for measurements (other categories remain unchanged). After setting the "acl.measurement.only-accessible-fragments" parameter to true, in order to have access to a single measurement parameter, you do not have to grant access to all fragments types that the measurement has. Only measurements that have a defined set of types are returned, and additional types not listed in the inventory role configuration are removed from the returned measurements.
```
Comment (SAG-jher) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:56]: Please menition how to set "acl.measurement.only-accessible-fragments" parameter

Comment (jkot-softwareag) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md:56]: for me whole this documentation is missing examples (full, with 2 measurements and 2 requests). If I wouldn't test it in the past, I would probably not understand this chapter.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 7]
```diff
@@ -47,9 +47,14 @@ Permissions are grouped into the following categories:
 
 Add a permission to the role by clicking the plus icon next to the desired category.
 
-In the **Type** field, specify a type to further restrict the type of data that this permission applies to. Access will be only granted to objects that contain the specified **Type**.
+In the **Type** field, specify a type to further restrict the type of data that this permission applies to. Access will be only granted to objects that contain
+only the specified **Type**. If the selected object contains more fragment types than those defined in the inventory roles configuration, in order to display it, they also must be added to the inventory role configuration.
```
[the remark this answers is not in the document]
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Change in some places - on UI it is still Type

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

**Review by jkot-softwareag — APPROVED** [review]

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 8]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
 
 Add a permission to the role by clicking the plus icon next to the desired category.
 
-In the **Type** field, specify a type to further restrict the type of data that this permission applies to. Access will be only granted to objects that contain the specified **Type**.
-
-For example, assume that your device sends measurements related to device management, such as "c8y&#95;SignalStrength", and actual production measurements. You want a user to only see the device management measurements. In this case, enter "c8y&#95;SignalStrength" as type. This will allow the user to only see measurements that contain the type "c8y&#95;SignalStrength". Note that the user will be able to see the entire measurement object including other types that may be part of the same measurement object.
+In the **Type** field, specify a fragments to further restrict the types of data that this permission applies to. Access will be only granted to objects that contain
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:50]:
```suggestion
In the **Type** field, specify a fragment to further restrict the types of data that this permission applies to. Access will only be granted to objects that contain
```

Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:50]: Is there any reason why this field is not labeled "Fragment type"?

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 9]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+exactly the specified **Fragment Types**. If the selected object contains more fragment types than those defined in the inventory roles configuration, in order to display it, they also must be added to the inventory role configuration.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:51]:
```suggestion
exactly the specified fragment types. If the selected object contains more fragment types than those defined in the inventory roles configuration, in order to display it, they also must be added to the inventory role configuration.
```

Comment (BeateRixen) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md:51]: As this is not the name of the field, it is confusing to format it like a field name.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 10]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+
+For example, assume that your device sends measurements related to device management, such as "c8y_SignalStrength" but the measurement itself also has "c8y_Temperature" which you are not interested in.
+For the selected device, there are also measurements containing only the c8y_Temperature fragment.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:54]:
```suggestion
For the selected device, there are also measurements containing only the "c8y_Temperature" fragment.
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 11]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+        "source": { "id": "2480300" },
+        "time": "2013-07-02T16:32:30.152+02:00",
+        "type": "SignalStrength",
+        "c8y_Temperature": {
+            "T": { "value": 10, "unit": "C" }
+        }
+    }
+
+
+You want a user to see only the device management measurements which have a fragment c8y_SignalStrength.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:82]:
```suggestion
You want a user to only see the device management measurements which have a fragment "c8y_SignalStrength". 
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 12]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+        "type": "SignalStrength",
+        "c8y_Temperature": {
+            "T": { "value": 10, "unit": "C" }
+        }
+    }
+
+
+You want a user to see only the device management measurements which have a fragment c8y_SignalStrength. 
+
+In the default configuration for inventory roles we need to provide access to all fragments that the measurement has, that is, "c8y_SignalStrength" and "c8y_Temperature".
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:84]:
```suggestion
In the default configuration for inventory roles we must provide access to all fragments that the measurement has, that is, "c8y_SignalStrength" and "c8y_Temperature".
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 13]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+
+<img src="/images/users-guide/Administration/default-roles-inventory.png" alt="Default inventory role configuration">
+
+Note that if a measurement also contains other fragments types, they must also be added in the inventory role configuration, and they also will be returned in the response.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:88]:
```suggestion
Note that if a measurement also contains other fragments types, they must also be added in the inventory role configuration, and they are also returned in the response. 
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 14]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+        "c8y_SignalStrength": {
+            "rssi": { "value": -53, "unit": "dBm" },
+            "ber": { "value": 0.14, "unit": "%" }
+        },
+        "c8y_Temperature": {
+            "T": { "value": 10, "unit": "C" }
+        }
+    }
+
+From version 10.17 a new tenant options **"acl.measurement.only-accessible-fragments"** in category "configuration" has been introduced for measurements (other APIs remain unchanged).
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:109]:
```suggestion
A tenant option `acl.measurement.only-accessible-fragments` in the category `configuration` can be used for measurements. 
```

Comment (BeateRixen) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md:109]: We usually don´t use phrases like this "was introduced in release xxx". We just add it to the documentation for the respective release. Is it relevant for the user when it was introduced otheriwse I would just describe it.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 15]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+    POST /tenant/options
+    ...
+    {
+        "category": "configuration",
+        "key": "acl.measurement.only-accessible-fragments",
+        "value": "true"
+    }
+
+
+After setting the parameter to true, in order to have access to a single measurement parameter like c8y_SignalStrength, you do not have to grant access to all fragments types that the measurement has.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:122]:
```suggestion
After setting the parameter to true, in order to have access to a single measurement fragment like "c8y_SignalStrength", you do not have to grant access to all fragments types that the measurement has.
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 16]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+
+For example, assume that your device sends measurements such as those in the previous example, including c8y_SignalStrength and c8y_Temperature and other measurements with c8y_Temperature only.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:124]:
```suggestion
For example, assume that your device sends measurements such as those in the previous example, including "c8y_SignalStrength" and "c8y_Temperature" and other measurements with "c8y_Temperature" only. 
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 17]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+        "time": "2013-07-02T16:32:30.152+02:00",
+        "type": "SignalStrength",
+        "c8y_Temperature": {
+            "T": { "value": 10, "unit": "C" }
+        }
+    }    
+
+    HTTP/1.1 201 Created
+
+You want a user to see only the device management measurements which have a fragment c8y_SignalStrength.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:153]:
```suggestion
You want a user to only see the device management measurements which have a fragment "c8y_SignalStrength".
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 8 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md:50]: I don't think so, maybe it should be changed.

Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 9 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 11 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 13 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 14 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 15 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 17 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 16 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 10 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 18]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+    }
+
+
+You want a user to only see the device management measurements which have a fragment "c8y_SignalStrength".
+
+In the default configuration for inventory roles we need to provide access to all fragments that the measurement has, that is, "c8y_SignalStrength" and "c8y_Temperature".
+
+<img src="/images/users-guide/Administration/default-roles-inventory.png" alt="Default inventory role configuration">
+
+Note that if a measurement also contains other fragments types, they must also be added in the inventory role configuration, and they are also returned in the response.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:88]:
```suggestion
Note that if a measurement also contains other fragment types, they must also be added in the inventory role configuration, and they are also returned in the response.
```

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md [hunk 19]
```diff
@@ -47,9 +47,132 @@ Permissions are grouped into the following categories:
...
+        "c8y_SignalStrength": {
+            "rssi": { "value": -53, "unit": "dBm" },
+            "ber": { "value": 0.14, "unit": "%" }
+        },
+        "c8y_Temperature": {
+            "T": { "value": 10, "unit": "C" }
+        }
+    }
+
+A tenant option `acl.measurement.only-accessible-fragments` in the category `configuration` can be used for measurements.
```
Comment (BeateRixen) [inline content/standard-tenant/managing-permissions-bundle/inventory-roles.md:109]:
```suggestion
The tenant option `acl.measurement.only-accessible-fragments` in the category `configuration` can be used for measurements.
```

**Comment by BeateRixen** [comment]
@pkepa7 3 minor suggestions left from my side, one old one and two new ones. If you commit them I will approve.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 19 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

File: content/standard-tenant/managing-permissions-bundle/inventory-roles.md (hunk 18 above)
Comment (pkepa7) [reply content/standard-tenant/managing-permissions-bundle/inventory-roles.md]: Changed.

**Review by BeateRixen — APPROVED** [review]

**Pull request merged**